core/src/main/java/io/confluent/rest/metrics/MetricsResourceMethodApplicationListener.java

```diff
@@ -536,12 +536,14 @@ private MetricsRequestEventListener(final Map<Method, RequestScopedMetrics> metr
       this.metrics = metrics;
       this.time = time;
       this.enableGlobalStatsRequestTags = enableGlobalStatsRequestTags;
+      // CIAM-2673: if an exception occur in a filter that runs before this method listener,
```

Comment:
An instance of this inner class get created for every request?

Also in situations where "MATCHING_START is never reached" ... RequestEvent.Type.FINISHED still get called?

Curious about alternatives... I wonder if maybe we just don't emit a metric  if `RequestEvent.Type.MATCHING_START` is never hit?

Reply:
Yeah we create a container for every request to perform the metrics calculations. https://eclipse-ee4j.github.io/jersey.github.io/apidocs/2.34/jersey/org/glassfish/jersey/server/monitoring/RequestEventListener.html

For RequestEventListener:

> This will register the instance for listening of request events for one particular request. Once the processing of the request is finished, the instance will be ignored by the Jersey runtime and not used for processing of further requests.

Specifically for request starting:

> void onEvent([RequestEvent](https://eclipse-ee4j.github.io/jersey.github.io/apidocs/2.34/jersey/org/glassfish/jersey/server/monitoring/RequestEvent.html) event)
>
>The method is called when new request event occurs. This method will never be called for method [RequestEvent.Type.START](https://eclipse-ee4j.github.io/jersey.github.io/apidocs/2.34/jersey/org/glassfish/jersey/server/monitoring/RequestEvent.Type.html#START) as this event is handled by [ApplicationEventListener](https://eclipse-ee4j.github.io/jersey.github.io/apidocs/2.34/jersey/org/glassfish/jersey/server/monitoring/ApplicationEventListener.html).

Reply:
Yeah its possible to not emit any metric if the resource is not matched.